aws-datastore/src/test/java/com/amplifyframework/datastore/appsync/ApiMocking.java

```diff
@@ -0,0 +1,150 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.doAnswer;
+
+/**
+ * Utility to mock behaviors of the API category.
+ */
+public final class ApiMocking {
+
+    private ApiMocking(){
+
+    }
+
+    /**
+     * Mock successful subscription start calls.
+     * @param mockApi Mock object of type {@link GraphQLBehavior}
+     */
+    public static void mockSubscriptionStart(GraphQLBehavior mockApi) {
+        doAnswer(invocation -> {
+            final int indexOfOnStart = 1;
+            Consumer<String> onStart = invocation.getArgument(indexOfOnStart);
+            onStart.accept(RandomString.string());
+            return null;
+        }).when(mockApi).subscribe(
+            any(), // Class<T>
+            any(), // Consumer<String>, onStart
+            any(), // Consumer<GraphQLResponse<ModelWithMetadata<T>>>, onNextResponse
+            any(), // Consumer<DataStoreException>, onSubscriptionFailure
+            any() // Action, onSubscriptionCompleted
+        );
+    }
+
+    /**
+     * Mock a successful mutation call made for a given model.
+     * @param mockApi Mock object of type {@link GraphQLBehavior}
+     * @param model The model for the mutation operation.
+     * @param <M> The model type.
+     */
+    public static <M extends Model> void mockSuccessfulMutation(GraphQLBehavior mockApi, M model) {
+        doAnswer(invocation -> {
+            // Simulate a successful response callback from the create() method.
+            final int indexOfModelBeingCreated = 0;
+            final int indexOfResultConsumer = 1;
+            GraphQLRequest<M> request = invocation.getArgument(indexOfModelBeingCreated);
+
+            // Pass back a ModelWithMetadata. Model is the one provided.
+            ModelMetadata metadata =
+                new ModelMetadata(model.getId(), false, 1, new Temporal.Timestamp());
+            ModelWithMetadata<M> modelWithMetadata = new ModelWithMetadata<>(model, metadata);
+            Consumer<GraphQLResponse<ModelWithMetadata<M>>> onResult =
+                invocation.getArgument(indexOfResultConsumer);
+            onResult.accept(new GraphQLResponse<>(modelWithMetadata, Collections.emptyList()));
+
+            // Technically, create() returns a Cancelable...
+            return null;
+        }).when(mockApi).mutate(
+            argThat(getMatcherFor(model)), // Match the id of the model passed in to the function.
+            any(), // onResponse
+            any() // onFailure
+        );
+    }
+
+    /**
+     * Mock the results of successful query operation triggered by the DataStore.
+     * @param mockApi Mock object of type {@link GraphQLBehavior}
+     * @param models A list of models to be returned in the result.
+     * @param <M> The model type.
+     */
+    @SuppressWarnings("unchecked")
+    public static <M extends Model> void mockSuccessfulSyncQuery(GraphQLBehavior mockApi, M... models) {
```

Comment:
This method name is leaking an abstraction, I think. The API category doesn't know about "sync"ing, that's a datastore thing. However, the class is called `ApiMocking`. To stay pure to its charter, this should probably not mention "sync" anywhere, just "query."

Also, since `ModelWithMetadata` is essentially a DataStore detail, I would pull that out of the implementation as well. Instead, you could provide them at the call site:
```java
public static <T> void mockSuccessfulSyncQuery(GraphQLBehavior mockApi, T... results) {
    ...
    // Implementation no longer mentions ModelWithMetadata either
    onResults.accept(new GraphQLResponse<PaginatedResult<T>>(
                new PaginatedResult<T>(syncQueryResults, null));
    ...
}
```